applications/tari_console_wallet/src/ui/state/app_state.rs

```diff
@@ -586,6 +586,12 @@ impl AppStateInner {
                     self.refresh_balance().await?;
                     self.updated = true;
                     return Ok(());
+                } else if (tx.status == TransactionStatus::Pending || tx.status == TransactionStatus::Completed) &&
```

Comment:
This is a needless repetition 
```Rust
    self.refresh_balance().await?;
    self.updated = true;
```
occur at the bottom of this function (Line 609) regardless of what happens here.